### ISSUE TITLE
LIFT: reflect our preferred directory structure 

### ISSUE THREAD

**Issue by obuckley-kenzan** [ThomsonReuters-IPS/angular-styleguide#1]
https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#locate

The convention we have chooses is as follows
```
/bower_components
/src
  /modules
   /bootstrap  //all app specific functionality, and the mounting module to bootstrap the app to the DOM 
     bootstrap-module.js
   /components  // directives
      /avengers
      /blocks
        /exception
        /logger
      /core
      /dashboard
      /data
      /layout
      /widgets
  /views  // routable controllers
```

cc: @paul-barry-kenzan / @franciov

**Comment by obuckley-kenzan** [comment]
In addition, we often tend to group a component by its parts, eg.
```
/componentA
  /controllers/
    controller-a.js
  /directives/
  /services/
  /etc/
```

To support LIFT, should we keep these component directories flatter?  At what point do we want to break out of this convention and eventually group?  Some examples for reference of different approaches so far

- https://github.com/ThomsonReuters-IPS/webui-component-authentication/tree/master/src/modules/authentication
- https://github.com/ThomsonReuters-IPS/webui-component-i18n/tree/master/src
- https://github.com/ThomsonReuters-IPS/webui-component-pubsub/tree/master/src/modules/pubsub-factory

**Comment by franciov** [comment]
The code style guide [suggests](https://github.com/ThomsonReuters-IPS/angular-styleguide/tree/master/a1#style-y143) to start _considering_ separation when you yo 7+ files.

> Keep a flat folder structure as long as possible. When you get to 7+ files, begin considering separation.
> 
> Why?: Nobody wants to search 7 levels of folders to find a file. Think about menus on web sites … anything deeper than 2 should take serious consideration. In a folder structure there is no hard and fast number rule, but when a folder has 7-10 files, that may be time to create subfolders. Base it on your comfort level. Use a flatter structure until there is an obvious value (to help the rest of LIFT) in creating a new folder.

Actually, I would say 7+ files _of the same type_: I think that even if we have 15+ files in the same level-1 folder, it's not an issue if we respect the [file naming conventions](https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#naming). In other words, I tend to favour the file name over the folder structure. As an example for a quite big (in terms of number of files) component like [search-result-types](https://github.com/ThomsonReuters-IPS/1p-frontend-new/tree/develop/1p-frontend-webapp/src/modules/components/search-result-types), I tend to favour this flat structure:

```
article-results.directive.js
article-results.directive.spec.js
article-results.html
patent-results.directive.js
patent-results.directive.spec.js
patent-results.html
people-results.directive.js
people-results.directive.spec.js
people-results.html
post-result.directive.js
post-results.directive.spec.js
post-results.html
search-result-type.directive.js
search-result-type.directive.spec.js
search-result-type.module.js
search-type.less
```

instead of the current structure:

```
directives
    article-results-directive.js
    article-results-directive.test.js
    patent-results-directive.js
    patent-results-directive.test.js
    people-result-directive.js
    people-results-directive.test.js
    post-result-directive.js
    post-results-directive.test.js
    search-result-type-directive.js
    search-result-type-directive.test.js
less
    search-type.less
templates
    article-results.html
    patent-results.html
    people-results.html
    post-results.html
search-result-type-module.js
```

Or, even better, a [folders-by-(sub)features](https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#folders-by-feature-structure) approach

```
article-results
    article-results.directive.js
    article-results.directive.spec.js
    article-results.html
patent-results
    patent-results.directive.js
    patent-results.directive.spec.js
    patent-results.html
people-results
    people-results.directive.js
    people-results.directive.spec.js
    people-results.html
post-result
    post-result.directive.js
    post-results.directive.spec.js
    post-results.html
search-result-type
    search-result-type.directive.js
    search-result-type.directive.spec.js
    search-type.less
search-result-type.module.js
```

**Comment by obuckley-kenzan** [comment]
thanks for the feedback @franciov .  So given that, do you then feel ok with layout as in webui-authentication-authentication?

**Comment by franciov** [comment]
@obuckley-kenzan Yes, [webui-component-authentication](https://github.com/ThomsonReuters-IPS/webui-component-authentication/tree/master/src/modules/authentication) has a flat structure and I tend to favour it.
